src/RecordManager/Finna/Record/Ead3.php

```diff
@@ -1367,4 +1371,48 @@ protected function isTimeIntervalNode(\SimpleXMLElement $node): bool
         return mb_strtolower((string)$node->attributes()->localtype, 'UTF-8')
             === self::RELATOR_TIME_INTERVAL;
     }
+
+    /**
+     * Get resource identifiers, used for identifier_txtP_mv and file_identifier_string_mv
+     *
+     * @return array<string,array> [ids, fileNames]
+     */
+    protected function getResourceIdentifiers(): array
+    {
+        $cacheKey = __FUNCTION__;
+        if ($this->resultCache[$cacheKey] ?? false) {
+            return $this->resultCache[$cacheKey];
+        }
+        $ids = [];
+        $fileNames = [];
+        $matchAttributes = function (\SimpleXMLElement $check) use (&$ids, &$fileNames) {
+            if ($check && $attrs = $check->attributes()) {
+                if ($identifier = trim((string)$attrs->identifier)) {
+                    $ids[] = $identifier;
+                }
+                if ($linktitle = trim((string)$attrs->linktitle)) {
+                    $fileNames[] = $linktitle;
+                }
+                if ($href = trim((string)$attrs->href)) {
+                    if (!preg_match('/^http(s)?:\/\//', $href)) {
+                        // Add scheme if missing
+                        $href = '://' . $href;
+                    }
+                    // We dont really need the domain and other parts from the url
+                    if ($path = parse_url($href, PHP_URL_PATH)) {
```

Comment:
Tässä on nyt pieni ongelma, kun URL:sta ei ihan näin helposti saa tiedoston nimeä kuin tietyissä tilanteissa. Jos vaikka URL on tyyliin `https://jotain/getDocument?doc=filename`, niin tuosta tulisi ´getDocument`, joka ei nyt ole kauhean hyödyllinen. Tai `https://jotain/docs/document/pdf`... 
Tai `https://docs.google.com/document/d/1rqeyEYI7g6uVjNo3gJtc8pY9-sNAYOqUWsxwV4JleGw/edit?usp=sharing`

Reply:
joo, aika hankalahan toi on tosta selvitellä. Mitä jos vedän vaan pois